### PR TITLE
pre-roll ads

### PR DESCRIPTION
**Pull request by seanyesmunt**

### CODE REVIEW

**Comment by seanyesmunt** [comment]
@DispatchCommit looking for feedback on this. Some of the real values are commented out to make it easier to test locally.

File: ui/component/viewers/videoViewer/view.jsx [hunk 1]
```diff
@@ -217,16 +268,50 @@ function VideoViewer(props: Props) {
...
+            </I18nMessage>
+          </span>
+        </>
+      )}
+
+      {!isFetchingAd && (
+        <VideoJs
+          adUrl={adUrl}
+          source={adUrl || source}
+          sourceType={forcePlayer || adUrl ? 'video/mp4' : contentType}
```
Comment (seanyesmunt) [inline ui/component/viewers/videoViewer/view.jsx:306]: Don't really like how I did the `source` and `sourceType` props

**Comment by DispatchCommit** [comment]
https://github.com/videojs/videojs-contrib-ads

this library would probably solve some issues

**Comment by CodeLikeCrazE** [comment]
If you are going to add ads, then at least make them able to be disabled by the user (and also if the user is signed in, just remove the ad, without a post to the iframe, because some people, like me, disable javascript in Google Ads).

**Comment by CodeLikeCrazE** [comment]
Also for something as big as adding ads, there almost definitely be more than 2 people saying "yeah okay that's fine".

**Pull request merged**